### PR TITLE
Seed Page

### PR DESCRIPTION
**Pull request by kupppo**
This PR introduces the Seed Page, a permalink for any generated seed. This URL can be shared anywhere and will generate the same seed. It is intended to be used with race generation bots and amongst users to play the same seed. The key goals are to make seeds instantly available to view their signatures, download without re-uploading the vanilla file, and then link to possible customizations.

## Overview
The Seed Page is located at `/seed.html` and reads from a number of query string parameters.
* `num` is the random number used in generating the seed.
* `mode` is the shortcode for the game mode used in logic (ie `mm` or `rm`).
* `download` is an optional parameter that will begin to automatically download the patched file if the vanilla file is available in browser. If this is not present, the file will not auto-download.

## Preview
Screenshot 2023-03-06 at 5 40 54 PM

## Notes
There are a few key aspects to this PR that are noteworthy:
* Titles were added to `game_modes` in`sm-rando.js` to allow for displaying the full name of the mode instead of its shortcode.
* Signatures have been added to `seed-page.js` to allow for looking up the signature of a seed. This data was pulled from [gameoptions.asm](https://github.com/dashrando/dash-template-asm/blob/main/src/fileselect/gameoptions.asm#L85-L117) in the [dash-template-asm](https://github.com/dashrando/dash-template-asm) repository.
* `RandomizeRom` and other functions involved in the seed rolling aspect were refactored to have as little reliance on the DOM as possible. This allows for the same seed to be rolled by the same parameters, regardless of the user interface present.
* `RandomizeRom` now accepts the following parameters:
  * `seed`: This is the random number used for generating the seed.
  * `game_mode`: This is the shortcode for the mode of game play such as `mm` or `rf`.
  * `opts`: Options for rolling the seed. These should be considered game specific parameters such as boss shuffle, area randomization or item fanfare.
  * `config`: Configurable options for non-gameplay related things. This would be for parameters like the vanilla file.
* The refactoring of `RandomizeRom` is a precursor to allow for rolling seeds in a non-browser environment, such as a Discord bot, racetime.gg bot or CLI.

## Testing
- [x] Load the page without required parameters
- [x] Load the page without vanilla file stored
- [x] Load the page with vanilla file stored
- [x] Load the page with vanilla file stored + download prefilled
- [x] Confirm generated seed is identical between seed page, dev generate page and production generate page

### CODE REVIEW

**Comment by MassHesteria** [comment]
Looks awesome 😃

Two requests:
1) Can you remove the "customize" div for now? We'll probably add it somewhere else later.
2) Could we prompt for the vanilla rom if it isn't loaded?

**Comment by MassHesteria** [comment]
One more thing: can we change NUMBER to SEED and change the `num` param to `seed`.

**Comment by kupppo** [comment]
No problem on all counts. Will do shortly.

This is what the page looks like when you haven't loaded a vanilla rom yet. This was done I just hadn't pushed yet.
Screenshot 2023-03-08 at 8 55 27 PM

**Comment by kupppo** [comment]
All changes should be done now. Let me know if you have any more feedback.

File: seed.css [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  color: #fff;
+  text-decoration-color: transparent;
+}
+
+#seed-footer {
+  margin: 40px 0 0;
+  font-size: 14px;
+  line-height: 18px;
+  opacity: 0;
+  transition: opacity 0.3s ease-in-out 0.8s;
```
Comment (MassHesteria) [inline seed.css:200]:
```suggestion
  transition: opacity 0.3s ease-in-out 0.6s;
```

Comment (kupppo) [reply seed.css]: Good catch. This has been pushed along with removing the `#customize` related CSS.

**Review by MassHesteria — APPROVED** [review]

**Pull request merged**